sdk/core/Azure.Core/tests/DynamicData/DynamicJsonSerializationTests.cs

```diff
@@ -0,0 +1,587 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
TODO: add relevant tests from this PR: https://github.com/Azure/azure-sdk-for-net/pull/36947